pkg/api/types.go

```diff
@@ -2143,6 +2143,20 @@ const (
 
 	// SSHAuthPrivateKey is the key of the required SSH private key for SecretTypeSSHAuth secrets
 	SSHAuthPrivateKey = "ssh-privatekey"
+
+	// SecretTypeTLS contains information about SSL termination used with the Ingress resource.
+	//
+	// Required fields:
+	// - Secret.Data["key"] - TLS private key.
+	//   Secret.Data["cert"] - TLS certificate.
+	//   Secret.Data["chain"] - optional certificate chain.
+	// TODO: Consider supporting different formats, specifying CA/destinationCA.
+	SecretTypeTLS SecretType = "kubernetes.io/tls"
```

Comment:
That was part of my motivation for including "Chain" and being all hand-wavy in the comments about what chain it was. By client cert I'm assuming you mean the cert a server would use to re-encrypt (or the cert a client would use for client auth), does they need to be different? 

Or were you talking about something from the client that might include a csr etc?